llama-index-core/llama_index/core/base/base_query_engine.py

```diff
@@ -16,25 +16,38 @@
 from llama_index.core.callbacks.base import CallbackManager
 from llama_index.core.prompts.mixin import PromptDictType, PromptMixin
 from llama_index.core.schema import NodeWithScore, QueryBundle, QueryType
-from llama_index.core.instrumentation import DispatcherSpanMixin
 from llama_index.core.instrumentation.events.query import (
     QueryEndEvent,
     QueryStartEvent,
 )
 import llama_index.core.instrumentation as instrument
+from llama_index.core.workflow.workflow import Workflow, _WorkflowMeta
+from llama_index.core.async_utils import asyncio_run
+from llama_index.core.callbacks.schema import CBEventType, EventPayload
+from llama_index.core.bridge.pydantic import BaseModel
 
 dispatcher = instrument.get_dispatcher(__name__)
 logger = logging.getLogger(__name__)
 
 
-class BaseQueryEngine(ChainableMixin, PromptMixin, DispatcherSpanMixin):
+PydanticMetaclass = type(BaseModel)
+
+
+class CombinedMeta(_WorkflowMeta, ABCMeta):
+    pass
+
+
+class BaseQueryEngine(ChainableMixin, PromptMixin, Workflow, metaclass=CombinedMeta):
     """Base query engine."""
 
     def __init__(
         self,
         callback_manager: Optional[CallbackManager],
+        timeout: float = 120.0,
+        **kwargs: Any,
     ) -> None:
         self.callback_manager = callback_manager or CallbackManager([])
+        Workflow.__init__(self, timeout=timeout, **kwargs)
```

Comment:
passing kwargs is a little dangerous, we should only pass in the args that workflow accepts

```python
class Workflow(metaclass=_WorkflowMeta):
    def __init__(
        self,
        timeout: Optional[float] = 10.0,
        disable_validation: bool = False,
        verbose: bool = False,
        service_manager: ServiceManager = ServiceManager(),
    ) -> None:
```